test/lib/request.lib.test.js

```diff
@@ -0,0 +1,189 @@
+      })
+
+      describe('the result it returns', () => {
+        it("has a 'succeeded' property marked as false", async () => {
+          const result = await RequestLib.get(testDomain)
+
+          expect(result.succeeded).to.be.false()
+        })
+
+        it("has a 'response' property containg the web response", async () => {
+          const result = await RequestLib.get(testDomain)
+
+          expect(result.response).to.exist()
+          expect(result.response.statusCode).to.equal(500)
+          expect(result.response.body).to.equal('{"data":"hello world"}')
+        })
+      })
+    })
+
+    describe('because there was a network issue', () => {
+      beforeEach(() => {
+        Nock(testDomain).get('/').replyWithError({ code: 'ECONNRESET' })
+      })
+
+      describe('the result it returns', () => {
+        it("has a 'succeeded' property marked as false", async () => {
+          const result = await RequestLib.get(testDomain)
+
+          expect(result.succeeded).to.be.false()
+        })
+
+        it("has a 'response' property containg the error", async () => {
+          const result = await RequestLib.get(testDomain)
+
+          expect(result.response).to.exist()
+          expect(result.response).to.be.an.error()
+          expect(result.response.code).to.equal('ECONNRESET')
+        })
+      })
+    })
+
+    describe('because request timed out', () => {
+      beforeEach(async () => {
+        // Set the timeout value to 50ms for these tests
+        Sinon.replace(requestConfig, 'timeout', 50)
+      })
+
+      // Because of the fake delay in this test, Lab will timeout (by default tests have 2 seconds to finish). So, we
+      // have to override the timeout for this specific test to all it to complete
+      describe('and all retries fail', { timeout: 5000 }, () => {
+        beforeEach(async () => {
+          Nock(testDomain)
+            .get(() => true)
+            .delay(100)
+            .reply(200, { data: 'hello world' })
+            .persist()
+        })
+
+        describe('the result it returns', () => {
+          it("has a 'succeeded' property marked as false", async () => {
+            const result = await RequestLib.get(testDomain)
+
+            expect(result.succeeded).to.be.false()
+          })
+
+          it("has a 'response' property containg the error", async () => {
+            const result = await RequestLib.get(testDomain)
+
+            expect(result.response).to.exist()
+            expect(result.response).to.be.an.error()
+            expect(result.response.code).to.equal('ETIMEDOUT')
+          })
+        })
+      })
+
+      describe('and a retry succeeds', () => {
+        beforeEach(async () => {
+          // The first response will time out, the second response will return OK
+          Nock(testDomain)
+            .get(() => true)
+            .delay(100)
+            .reply(200, { data: 'hello world' })
```

Comment:
```suggestion
            .reply(200, { data: 'delayed hello world' })
```
With the data the same for both the delayed and rapid replies, if the delayed reply wasn't getting timed out the test would still pass